### PR TITLE
AL/MIN-253. User mnt balance.

### PR DESCRIPTION
**Pull request by lut1k**
**Description:**

Implemented RPC `get_unclaimed_mnt_balance()` for paller mnt-token. This RPC mets MNT accrued but not yet transferred to user.
Implemented tests for this RPC.

**Changes in storage or API:**

Added a description of the RPC to the API documentation.

**Checklist**

- [ ] Code is covered with tests
- [ ] Code is covered with benchmarks
- [ ] Fresh benchmark results included

### CODE REVIEW

File: pallets/mnt-token/rpc/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+/// A struct that implements the [`MntTokenApi`].
+pub struct MntToken<C, B> {
+	client: Arc<C>,
+	_marker: std::marker::PhantomData<B>,
+}
+
+impl<C, B> MntToken<C, B> {
+	/// Create new `LiquidityPool` with the given reference to the client.
```
Comment (kurotychb) [inline pallets/mnt-token/rpc/src/lib.rs:26]: There is no LiquidityPool

Comment (lut1k) [reply pallets/mnt-token/rpc/src/lib.rs:26]: Fixed

File: pallets/mnt-token/src/lib.rs [hunk 2]
```diff
@@ -361,6 +363,23 @@ impl<T: Config> Pallet<T> {
...
+impl<T: Config> Pallet<T> {
+	/// Gets MNT accrued but not yet transferred to user.
+	pub fn get_unclaimed_mnt_balance(account_id: &T::AccountId) -> BalanceResult {
+		let accrued_mnt =
+			MntSpeeds::<T>::iter().try_fold(Balance::zero(), |current_accrued, (pool_id, _)| -> BalanceResult {
+				let (mut accrued_borrow_mnt, mut accrued_supply_mnt) = (Balance::zero(), Balance::zero());
+				Self::update_mnt_borrow_index(pool_id)?;
+				accrued_borrow_mnt += Self::distribute_borrower_mnt(pool_id, account_id, true)?;
+				Self::update_mnt_supply_index(pool_id)?;
+				accrued_supply_mnt += Self::distribute_supplier_mnt(pool_id, account_id, true)?;
```
Comment (dev-vadim) [inline pallets/mnt-token/src/lib.rs:376]: Can we update storage during an rpc call?

Comment (lut1k) [reply pallets/mnt-token/src/lib.rs:376]: The RPC does not change the storage in DB

Comment (dev-vadim) [inline pallets/mnt-token/src/lib.rs:376]:
```suggestion
				Self::update_mnt_borrow_index(pool_id)?;
				let accrued_borrow_mnt = Self::distribute_borrower_mnt(pool_id, account_id, true)?;
				Self::update_mnt_supply_index(pool_id)?;
				let accrued_supply_mnt = Self::distribute_supplier_mnt(pool_id, account_id, true)?;
```

Comment (lut1k) [reply pallets/mnt-token/src/lib.rs:376]: Fixed

**Review by dev-vadim — DISMISSED** [review]

**Review by Apollo50 — APPROVED** [review]

File: runtime/src/tests/rpc_tests.rs [hunk 3]
```diff
@@ -685,65 +657,56 @@ fn get_user_total_collateral_rpc_should_work() {
...
+				ETH => {
+					assert_eq!(price, Some(Price::saturating_from_integer(10_000)));
+				}
+				BTC => {
+					assert_eq!(price, Some(Price::saturating_from_integer(10_000)));
 				}
+				KSM => {
+					assert_eq!(price, Some(Price::saturating_from_integer(10_000)));
+				}
+				_ => panic!("Unexpected token!"),
```
Comment (kurotychb) [inline runtime/src/tests/rpc_tests.rs:706]: You can write it more compact
```
match currency_id {
  DOT => assert_eq!(price, None),
  ETH | BTC | KSM => assert_eq!(price, Some(Price::saturating_from_integer(10_000)))
  _ => panic!("Unexpected token!"),
}
```

Comment (lut1k) [reply runtime/src/tests/rpc_tests.rs:706]: Fixed >.<

**Review by ales-by — APPROVED** [review]

**Review by kurotychb — APPROVED** [review]

**Pull request merged**